src/server/status_server.rs

```diff
@@ -537,6 +549,79 @@ impl StatusServer {
     pub fn listening_addr(&self) -> SocketAddr {
         self.addr.unwrap()
     }
+
+    fn register_addr(&self, status_addr: String) {
+        if self.pd_client.is_none() {
+            return;
+        }
+        let pd_client = self.pd_client.as_ref().unwrap();
+        let client = Client::new();
+        let json = {
+            let mut body = std::collections::HashMap::new();
+            body.insert("component".to_owned(), COMPONENT.to_owned());
+            body.insert("addr".to_owned(), status_addr);
+            serde_json::to_string(&body).unwrap()
+        };
+        for _ in 0..COMPONENT_REQUEST_RETRY {
+            for pd_addr in pd_client.get_leader().get_client_urls() {
+                let pd_addr = pd_addr
+                    .trim_start_matches("http://")
```

Comment:
How about using `Url::parse`?

Reply:
Is the client urls return from pd always starts with scheme like `http://` or `https://`?